wandb/__init__.py

```diff
@@ -287,21 +288,31 @@ def _init_jupyter(run):
 
     api = InternalApi()
     if not api.api_key:
-        termerror(
-            "Not authenticated.  Copy a key from https://app.wandb.ai/authorize")
-        key = getpass.getpass("API Key: ").strip()
-        if len(key) == 40:
-            os.environ[env.API_KEY] = key
-            util.write_netrc(api.api_url, "user", key)
-        else:
-            raise ValueError("API Key must be 40 characters long")
+        key = None
+        if 'google.colab' in sys.modules:
+            key = jupyter.attempt_colab_login(api.app_url)
+            if key:
+                os.environ[env.API_KEY] = key
+                util.write_netrc(api.api_url, "user", key)
+        if not key:
+            termerror(
+                "Not authenticated.  Copy a key from https://app.wandb.ai/authorize")
+            key = getpass.getpass("API Key: ").strip()
+            if len(key) == 40:
+                os.environ[env.API_KEY] = key
+                util.write_netrc(api.api_url, "user", key)
+            else:
+                raise ValueError("API Key must be 40 characters long")
         # Ensure our api client picks up the new key
         api = InternalApi()
+        run._api = api
```

Comment:
Would it be better to use the API from the Run? Does it already have some settings initialized that it would be good to preserve?

Reply:
I refactored this to call api.reauth and got rid of passing api to RunManager